crates/bevy_ui/src/lib.rs

```diff
@@ -68,28 +80,33 @@ pub mod prelude {
 
 use bevy_app::prelude::*;
 use bevy_ecs::prelude::*;
+#[cfg(feature = "bevy_render")]
 use bevy_input::InputSystem;
+#[cfg(feature = "bevy_render")]
 use bevy_render::{
     camera::CameraUpdateSystem,
     view::{check_visibility, VisibilitySystems},
     RenderApp,
 };
 use bevy_transform::TransformSystem;
-use layout::ui_surface::UiSurface;
 use stack::ui_stack_system;
 pub use stack::UiStack;
-use update::{update_clipping_system, update_target_camera_system};
+use update::update_clipping_system;
+#[cfg(feature = "bevy_render")]
+use update::update_target_camera_system;
 
 /// The basic plugin for Bevy UI
 pub struct UiPlugin {
     /// If set to false, the UI's rendering systems won't be added to the `RenderApp` and no UI elements will be drawn.
     /// The layout and interaction components will still be updated as normal.
+    #[cfg(feature = "bevy_render")]
     pub enable_rendering: bool,
 }
 
 impl Default for UiPlugin {
     fn default() -> Self {
         Self {
+            #[cfg(feature = "bevy_render")]
             enable_rendering: true,
         }
     }
```

Comment:
```suggestion
/// The basic plugin for Bevy UI
#[derive(Default)]
pub struct UiPlugin;
```

Shouldn't be needed anymore.